### DISCUSSION TITLE
HTTP CLIENT failing

### DISCUSSION THREAD

**Discussion by IamVishalSharma** [open-telemetry/opentelemetry-cpp#2375] [General]
When I am testing the sending instruments of my c++ app with http client to collector docker in my local lab.  all was working fine.

when the same app running as pod in K8s Cluster sending to a collector docker only , ending in the export error.  I am not able to understand the error. and for which instrument it is failing. So Clueless how to debug further as in my local setup evey thing works fine.

[Error] File: /home/test/otel/opentelemetry-cpp/exporters/otlp/src/otlp_http_client.cc:103 [OTLP HTTP Client] Export failed, Status:400, Header:	Content-Length: 251
,	Content-Type: application/json
,	Date: Wed, 18 Oct 2023 07:29:01 GMT
,Body:{"code":3,"message":"ReadUint64: unsupported value type, error found in #10 byte of ...|asDouble\":null,\"star|..., bigger context ...|gregationTemporality\":2,\"dataPoints\":[{\"asDouble\":null,\"startTimeUnixNano\":\"1697612761964413404\",\"ti|..."}

[Error] File: /home/test/otel/opentelemetry-cpp/exporters/otlp/src/otlp_http_metric_exporter.cc:126 [OTLP HTTP Client] ERROR: Export 5 metric(s) error: 1

**Comment by IamVishalSharma** [comment]
@marcalff could you please help

**Comment by lalitb** [comment]
`|asDouble":null` - this seems wrong. The metric data point shouldn't be null.

**Reply by IamVishalSharma** [reply]
@lalitb  thank you very much.  This metric data point is the value or can be data point attribute as well. and how to figure out for which instrument it is happening

**Reply by lalitb** [reply]
It is supposed to be value, not the attributes. Difficult to find the right instrument from the error message. You may need to debug it further by enabling them one at a time and see which one errors out. And share the code for that instrument here. 
Also see if you can run collector container in verbose mode, in case that provides more details.

**Comment by marcalff** [comment]
Not familiar with the error messages returned from the collector, so this makes it hard to debug.

One thing that stands out is seeing `ReadUint64` next to `asDouble` ...

Please double check each uses of variants, to make sure that Uint64 counters are assigned an int value, not a double value.

If this turns out to be the case, please let us know, as opentelemetry-cpp needs to be more robust and detect this instead of passing bad data down the chain.
